test/fika/type_checker_test.exs

```diff
@@ -233,39 +233,51 @@ defmodule Fika.TypeCheckerTest do
 
       ast = TestParser.expression!(str)
 
-      assert {:ok, "{Int,Int,Int}", _env} = TypeChecker.infer_exp(Env.init(), ast)
+      assert {:ok, %T.Tuple{elements: %T.ArgList{value: [:Int, :Int, :Int]}}, _env} =
+               TypeChecker.infer_exp(Env.init(), ast)
     end
 
     test "tuple of integers and floats" do
       str = "{1, 2/3, 3}"
 
       ast = TestParser.expression!(str)
 
-      assert {:ok, "{Int,Float,Int}", _env} = TypeChecker.infer_exp(Env.init(), ast)
+      assert {:ok, %T.Tuple{elements: %T.ArgList{value: [:Int, :Float, :Int]}}, _env} =
+               TypeChecker.infer_exp(Env.init(), ast)
     end
 
     test "tuple of floats inferred from fn calls" do
       str = "{1/2, 2/3}"
 
       ast = TestParser.expression!(str)
 
-      assert {:ok, "{Float,Float}", _env} = TypeChecker.infer_exp(Env.init(), ast)
+      assert {:ok, %T.Tuple{elements: %T.ArgList{value: [:Float, :Float]}}, _env} =
+               TypeChecker.infer_exp(Env.init(), ast)
     end
 
     test "tuple of strings" do
       str = ~s({"foo", "bar"})
 
       ast = TestParser.expression!(str)
 
-      assert {:ok, "{String,String}", _env} = TypeChecker.infer_exp(Env.init(), ast)
+      assert {:ok, %T.Tuple{elements: %T.ArgList{value: [:String, :String]}}, _env} =
+               TypeChecker.infer_exp(Env.init(), ast)
     end
 
     test "tuple of tuple of mixed types" do
       str = ~s({{1, 2/5}, {"3", true}})
 
       ast = TestParser.expression!(str)
 
-      assert {:ok, "{{Int,Float},{String,Bool}}", _env} = TypeChecker.infer_exp(Env.init(), ast)
+      assert {:ok,
+              %T.Tuple{
+                elements: %T.ArgList{
+                  value: [
+                    %T.Tuple{elements: %T.ArgList{value: [:Bool, :String]}},
+                    %T.Tuple{elements: %T.ArgList{value: [:Float, :Int]}}
+                  ]
+                }
+              }, _env} = TypeChecker.infer_exp(Env.init(), ast)
```

Comment:
The order of elements is important here - looks like the order needs to be reversed.

Reply:
Even though we decided to use ArgList struct, it looks like it doesn't give us much advantage. If we use simple lists instead of `%ArgList{value: []}`, we would end up writing `Enum.join(types, ", ")` in the to_string implementations of these structs but the advantage is that the type structs become leaner in the AST. What do you think?

Reply:
I'll look into what changes the ArgList really helps with. I suspect it's one of those things that made sense at first when everything was still string-dependent at the beggining of the PR